tests/core/test_device_save_loader.py

```diff
@@ -0,0 +1,197 @@
+from enum import Enum
+from os import path
+from typing import Any, Dict, List
+
+import numpy as np
+import numpy.typing as npt
+import pytest
+import yaml
+from bluesky.run_engine import RunEngine
+
+from ophyd_async.core import Device, SignalR, SignalRW
+from ophyd_async.core.device_save_loader import (
+    get_signal_values,
+    save_to_yaml,
+    walk_rw_signals,
+)
+from ophyd_async.epics.signal import epics_signal_r, epics_signal_rw
+
+
+class DummyChildDevice(Device):
+    def __init__(self) -> None:
+        self.sig1: SignalRW = epics_signal_rw(str, "Value1")
+        self.sig2: SignalR = epics_signal_r(str, "Value2")
+
+
+class EnumTest(str, Enum):
+    VAL1 = "val1"
+    VAL2 = "val2"
+
+
+class DummyDeviceGroup(Device):
+    def __init__(self, name: str):
+        self.child1: DummyChildDevice = DummyChildDevice()
+        self.child2: DummyChildDevice = DummyChildDevice()
+        self.parent_sig1: SignalRW = epics_signal_rw(str, "ParentValue1")
+        self.parent_sig2: SignalR = epics_signal_r(
+            int, "ParentValue2"
+        )  # Ensure only RW are found
+        self.parent_sig3: SignalRW = epics_signal_rw(str, "ParentValue3")
+        self.position: npt.NDArray[np.int32]
+
+
+@pytest.fixture
+async def device() -> DummyDeviceGroup:
+    device = DummyDeviceGroup("parent")
+    await device.connect(sim=True)
+    return device
+
+
+@pytest.fixture
+async def device_with_phases() -> DummyDeviceGroup:
+    device = DummyDeviceGroup("parent")
+    await device.connect(sim=True)
+
+    # Dummy function to check different phases save properly
+    def sort_signal_by_phase(self, values: Dict[str, Any]) -> List[Dict[str, Any]]:
+        phase_1 = {"child1.sig1": values["child1.sig1"]}
+        phase_2 = {"child2.sig1": values["child2.sig1"]}
+        return [phase_1, phase_2]
+
+    setattr(device, "sort_signal_by_phase", sort_signal_by_phase)
+    return device
+
+
+async def test_enum_yaml_formatting(tmp_path):
+    enums = [EnumTest(EnumTest.VAL1), EnumTest(EnumTest.VAL2)]
+    assert isinstance(enums[0], EnumTest)
+    save_to_yaml(enums, path.join(tmp_path, "test_file.yaml"))
+    with open(path.join(tmp_path, "test_file.yaml"), "r") as file:
+        yaml_content = yaml.load(file, yaml.Loader)
+        assert isinstance(yaml_content, list)
+        assert yaml_content[0] == EnumTest.VAL1
+        assert yaml_content[1] == EnumTest.VAL2
```

Comment:
these assertions all make sense, but I'd shorten them for readability:

```suggestion
    save_to_yaml(enums, path.join(tmp_path, "test_file.yaml"))
    with open(path.join(tmp_path, "test_file.yaml"), "r") as file:
        saved_enums = yaml.load(file, yaml.Loader)
    assert saved_enums == enums
```

This effectively combines the assert statements you currently have on lines 71-73, and removes the assert statement on 67 since that will never raise an AssertionError unless there is something very wrong with python or someone's redefined EnumTest... Up to you if you want to rename yaml_content or not

Reply:
```
async def test_enum_yaml_formatting(tmp_path):
    enums = [EnumTest.VAL1, EnumTest.VAL2]
    save_to_yaml(enums, path.join(tmp_path, "test_file.yaml"))
    with open(path.join(tmp_path, "test_file.yaml"), "r") as file:
        saved_enums = yaml.load(file, yaml.Loader)
    # check that save/load reduces from enum to str
    assert all(isinstance(value, str) for value in saved_enums)
    # check values of enums same
    assert saved_enums == enums
```
How about this? I'd like to have something that shows that reduction to a string when saving to yaml